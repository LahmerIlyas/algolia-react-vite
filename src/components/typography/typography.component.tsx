import classNames from 'classnames'

const styles = {
  s1: 'text-s1',
  s2: 'text-s2',
  s3: 'text-s3',
  s4: 'text-s4',
  s5: 'text-s5',
  s6: 'text-s6',
  btn: 'text-btn',
  lead: 'text-lead',
  body: 'text-body',
  caption: 'text-caption',
} as const

interface TypographyProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement> {
  variant: keyof typeof styles
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'a'
  className?: string
}

export const Typography: React.FC<TypographyProps> = ({ variant, className, children, ...other }) => {
  const c = classNames(styles[variant], {
    [className]: !!className,
  })

  return (
    <other.as className={c} {...other}>
      {children}
    </other.as>
  )
}

export default Typography
