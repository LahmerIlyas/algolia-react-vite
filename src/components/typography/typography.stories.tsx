import React from 'react'
import type { ComponentMeta, ComponentStory } from '@storybook/react'

import Typography from './typography.component'

export default {
  title: 'Example/Typography',
  component: Typography,
  argTypes: {
    variant: { control: 'color' },
  },
} as ComponentMeta<typeof Typography >

const Template: ComponentStory<typeof Typography> = () => (
  <React.Fragment>
    <Typography variant="s1" as="h1">
      Header 1
    </Typography>
    <Typography variant="s2" as="h2">
      Header 2
    </Typography>
    <Typography variant="s3" as="h3">
      Header 3
    </Typography>
    <Typography variant="s4" as="h4">
      Header 4
    </Typography>
    <Typography variant="s5" as="h5">
      Header 5
    </Typography>
    <Typography variant="s6" as="h6">
      Header 6
    </Typography>
  </React.Fragment>
)

export const ExampleStory = Template.bind({})
ExampleStory.args = {
}
