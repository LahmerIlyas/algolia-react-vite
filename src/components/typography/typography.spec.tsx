import { describe, expect, it } from 'vitest'
import renderer from 'react-test-renderer'
import { Typography } from './typography.component'

describe('Given typography element', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<Typography as="h1" variant="s1">H1</Typography>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
