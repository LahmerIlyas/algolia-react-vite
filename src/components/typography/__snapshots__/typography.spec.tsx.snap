// Vitest Snapshot v1

exports[`Given typography element > should renders correctly 1`] = `
<h1
  as="h1"
  className="text-s1"
>
  H1
</h1>
`;

exports[`Hi > renders correctly when there are no items 1`] = `
<h1
  as="h1"
  className="text-s1"
>
  H1
</h1>
`;
